packages/driver/src/cypress/error_messages.ts

```diff
@@ -1393,16 +1393,20 @@ export default {
       message: `${cmd('select')} was called with an array of arguments but does not have a \`multiple\` attribute set.`,
       docsUrl: 'https://on.cypress.io/select',
     },
+    invalid_number: {
+      message: `${cmd('select')} was called with an invalid index: \`{{index}}\`. Index must be a non-negative integer.`,
+      docsUrl: 'https://on.cypress.io/select',
+    },
     multiple_elements: {
       message: `${cmd('select')} can only be called on a single \`<select>\`. Your subject contained {{num}} elements.`,
       docsUrl: 'https://on.cypress.io/select',
     },
     multiple_matches: {
-      message: `${cmd('select')} matched more than one \`option\` by value or text: \`{{value}}\``,
+      message: `${cmd('select')} matched more than one \`option\` by value, index, or text: \`{{value}}\``,
```

Comment:
I actually don't think this is possible, for it to find multiple matches of the same index. 

```suggestion
      message: `${cmd('select')} matched more than one \`option\` by value, or text: \`{{value}}\``,
```